### DISCUSSION TITLE
Modify original graphical assets?

### DISCUSSION THREAD

**Discussion by Mr-Bajs** [ihhub/fheroes2#8246] [General]
Discovering that some terrains doesn't have abandoned mines. Then I started to wonder is it fine using parts of the copyrighted original assets to modify to create new?

For example these sort of assets ok to add?

Abandoned mine swamp
![abadonmineswampwosh](https://github.com/ihhub/fheroes2/assets/93934125/649ae400-8cfa-4479-992f-70e71b6d78cf)

Halfling dwelling swamp
![haflingdwellingswamp](https://github.com/ihhub/fheroes2/assets/93934125/1aee1e9d-1828-4b40-b782-21ea93825ab8)

Abandoned mine desert
![abandonedminedesert](https://github.com/ihhub/fheroes2/assets/93934125/91eaebf7-4abd-4849-8daf-f5a7c9660b34)

**Comment by ihhub** [comment]
If these objects are created by the engine during runtime then yes.

**Reply by Mr-Bajs** [reply]
Okey, so how does that work for the assets that are "modified" now? Is it an modified layer that shows on top of the original asset?

**Reply by ihhub** [reply]
We generate needed parts of images using existing assets and a set of image processing functions.

**Reply by Mr-Bajs** [reply]
So for example if you would want to create a dirt variant of the lookout tower, with dirt grass instead of green grass. Would providing just an transparent grass image that writes ontop ot the original image be an easy way to do it for some assets?

![OriginalTower](https://github.com/ihhub/fheroes2/assets/93934125/c568cb4f-7905-4f72-971d-7a0c30af8ed2)
![TowerDirtGrass](https://github.com/ihhub/fheroes2/assets/93934125/7514610d-0642-4153-b616-1e74b9eb37bd)
![DirtTower](https://github.com/ihhub/fheroes2/assets/93934125/3ac3c5c3-3e27-4010-aa53-91cb72b3680d)
